### PR TITLE
Begin properly implementing reparsing

### PR DESCRIPTION
**Pull request by CodaFi**
Code is still a disaster (hence WIP) but it passes the tests.

### CODE REVIEW

**Comment by CodaFi** [comment]
Some uncontroversial parts of this will be split off into smaller PRs.

File: Tests/ScopeCheck/ambiguous-operators.silt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  | true  : Bool
+
+if_then_else_ : Bool -> {A : Type} -> A -> A -> A
+if true  then x else y = x
+if false then x else y = y
+
+infix 1 if_then_else_
+
+_&&_ : Bool -> Bool -> Bool
+x && y = if_then_else_ x Bool y false
```
Comment (harlanhaskins) [inline Tests/ScopeCheck/ambiguous-operators.silt:21]: Any reason this call doesn’t use mixfix notation?

**Review by harlanhaskins — APPROVED** [review]

File: Sources/Lithosphere/SyntaxNodes.swift [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /// SyntaxNodes.swift
 /// Automatically generated by SyntaxGen. Do not edit!
 ///
-/// Copyright 2017-2018, The Silt Language Project.
+/// Copyright 2018, The Silt Language Project.
```
Comment (harlanhaskins) [inline Sources/Lithosphere/SyntaxNodes.swift]: Can you update SyntaxGen to put the right header in?

File: Tests/ScopeCheck/ambiguous-operators.silt (hunk 1 above)
Comment (CodaFi) [reply Tests/ScopeCheck/ambiguous-operators.silt:21]: Some of this tests early-exit.  I'll add more docs, but basically it's less expensive to traverse all the operators and filter them from the grammar if they don't appear in the expression than it is to build and parse the full grammar.  It's even less expensive to do no work at all if there are no operators!

File: Sources/Moho/Notation.swift [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+  /// Converts a fixity declaration to an internal fixity.
+  var fixity: Fixity {
+    let fixity: Fixity
+    switch self {
+    case let fds as NonFixDeclSyntax:
+      guard
+        case let .identifier(num) = fds.precedence.tokenKind,
+        let prec = Int(num, radix: 10)
+        else {
+          fatalError()
```
Comment (harlanhaskins) [inline Sources/Moho/Notation.swift]: Diagnostic here.

Comment (harlanhaskins) [reply Sources/Moho/Notation.swift]: And below

**Comment by CodaFi** [comment]
Great success!

⛵️

**Pull request merged**